src/kokkos/test/kokkos/TrackingRecHit2D_t.cc

```diff
@@ -0,0 +1,70 @@
+#include "KokkosCore/kokkosConfigCommon.h"
+#include "KokkosCore/kokkosConfig.h"
+
+#include "CondFormats/pixelCPEforGPU.h"
+#include "KokkosDataFormats/TrackingRecHit2DKokkos.h"
+
+namespace testTrackingRecHit2DKokkos {
+
+  template<typename MemorySpace>
+  void fill(Kokkos::View<TrackingRecHit2DSOAView, MemorySpace> hits) {
+
+    assert(hits.data());
+    auto &hits_ = *hits.data();
+
+    Kokkos::parallel_for("fill", Kokkos::RangePolicy<KokkosExecSpace>(0, 1024), KOKKOS_LAMBDA(const size_t i) {
+              assert(hits_.nHits() == 200);
+              if (i > 200) return;
+            });
+
+    return;
+  }
+
+  template<typename MemorySpace>
+  void verify(Kokkos::View<TrackingRecHit2DSOAView, MemorySpace> hits) {
+
+    assert(hits.data());
+
+    auto const &hits_ = *hits.data();
+
+    Kokkos::parallel_for("fill", Kokkos::RangePolicy<KokkosExecSpace>(0, 1024), KOKKOS_LAMBDA(const size_t i) {
+              assert(hits_.nHits() == 200);
+              if (i > 200) return;
+            });
+
+    return;
+  }
+
+  template<typename MemorySpace>
+  void runKernels(Kokkos::View<TrackingRecHit2DSOAView, MemorySpace> hits) {
+    assert(hits.data());
+
+    fill(hits);
+    verify(hits);    
+  }
+}  // namespace testTrackingRecHit2DKokkos
+
+namespace testTrackingRecHit2DKokkos {
+  template<typename MemorySpace>  
+  void runKernels(Kokkos::View<TrackingRecHit2DSOAView, MemorySpace> hits);
```

Comment:
Just to note that this declaration looks unnecessary, but it is in the CUDA version as well (who knows why), so it can stay here as well.

Reply:
I wondered about this point. Agree that this should be updated synchronously.